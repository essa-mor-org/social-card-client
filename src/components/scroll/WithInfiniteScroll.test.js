import React, { Component } from 'react';
import withInfiniteScroll from './WithInfiniteScroll';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';

class MockListComponent extends Component {
    render() {
        return (<div>Fake List</div>);
    }
}

let InfiniteScrollComponent = withInfiniteScroll(MockListComponent, { fetchMethodName: 'fetch' });

describe('withInfiniteScroll test', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(<InfiniteScrollComponent hasMore={false} fetch={jest.fn()} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
