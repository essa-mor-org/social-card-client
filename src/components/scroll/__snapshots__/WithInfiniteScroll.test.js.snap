// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withInfiniteScroll test renders correctly 1`] = `
<div>
  <div>
    Fake List
  </div>
</div>
`;
